src/main/java/org/apache/maven/buildcache/RemoteCacheRepositoryProvider.java

```diff
@@ -42,8 +43,12 @@
     public RemoteCacheRepositoryProvider( CacheConfig config, PlexusContainer container )
             throws ComponentLookupException
     {
-        config.initialize();
-        String hint = config.getTransport();
+        String hint = "noop";
+        if ( config.initialize() == CacheState.INITIALIZED )
```

Comment:
maybe implement DisabledCacheConfig or NoopCacheConfig to remove such checks across code?

Reply:
Agree, let's do in a next pr. This one is the hotfix.